### PR TITLE
voiceFirstUI settings update

### PR DESCRIPTION
**Pull request by vishnu-ravi**
for voiceFirstUI instead of added the assitantImage, we open the chat immediately and also a class added to iframe container to handle the css style for voice first UI

### CODE REVIEW

File: src/js/yeldaWebchatInjector.js [hunk 1]
```diff
@@ -158,25 +159,37 @@ class YeldaChat {
...
 
-      if (!hasCustomStyle || !customImage) {
-        this.webChatContainer.appendChild(this.assistantImage)
-        return
-      }
+      /**
+       * when isVoiceFirstUI is true, instead of adding the assistant image open the chat here,
+       * website-chat will take care rendering the voice first UI
+       */
+      if (!data.isDemo && isVoiceFirstUI) {
```
Comment (meelie) [inline src/js/yeldaWebchatInjector.js:170]: why (!data.isDemo) ? isDemo has the right to have isVoiceFirstUI
Can you explain ?

File: src/js/yeldaWebchatInjector.js [hunk 2]
```diff
@@ -158,25 +159,37 @@ class YeldaChat {
...
+        // voiceFirstUI added to the iframeContainer to remove box-shadow css style
+        // other styles can be added for voice first UI based on this class in the future if needed
+        this.iframeContainer.classList.add('voiceFirstUI')
+        this.openChat()
+      } else {
```
Comment (meelie) [inline src/js/yeldaWebchatInjector.js:175]: don't we need a return here ? if we don't the see the assistant image on mobile, no need to load it, right ?

File: src/js/yeldaWebchatInjector.js [hunk 3]
```diff
@@ -158,25 +159,37 @@ class YeldaChat {
...
-        this.webChatContainer.appendChild(this.assistantImage)
-        return
-      }
+      /**
+       * when isVoiceFirstUI is true, instead of adding the assistant image open the chat here,
+       * website-chat will take care rendering the voice first UI
+       */
+      if (!data.isDemo && isVoiceFirstUI) {
+        // voiceFirstUI added to the iframeContainer to remove box-shadow css style
+        // other styles can be added for voice first UI based on this class in the future if needed
```
Comment (meelie) [inline src/js/yeldaWebchatInjector.js:172]: To 100% honest I don't fully understand the English here

     /**
       * in isVoiceFirstUI mode
       * - the assistant should be opened directly and will never be closed 
       *    => we call directly openChat and do not add the assistant image
       * - we don't want the box-shadow css style 
       *    => we add voiceFirstUI to iframeContainer
       * - /chat vue render the voice first UI
       *   => nothing more to do here
       */

**Review by meelie — CHANGES_REQUESTED** [review]
thanks

File: src/js/yeldaWebchatInjector.js (hunk 1 above)
Comment (vishnu-ravi) [reply src/js/yeldaWebchatInjector.js:170]: I thought we don't show the voice first ui now. that's why, now I have removed it

Comment (meelie) [reply src/js/yeldaWebchatInjector.js]: Default value is false, so nobody is supposed to see it yet (excepting if they turn it on)

File: src/js/yeldaWebchatInjector.js [hunk 4]
```diff
@@ -158,25 +158,43 @@ class YeldaChat {
...
+       * => we call directly openChat and do not add the assistant image
+       * - we don't want the box-shadow css style
+       *    => we add voiceFirstUI to iframeContainer
+       * - /chat vue render the voice first UI
+       *   => nothing more to do here
+       */
+      if (isVoiceFirstUI) {
+        // voiceFirstUI added to the iframeContainer to remove box-shadow css style
+        // other styles can be added for voice first UI based on this class in the future if needed
+        this.iframeContainer.classList.add('voiceFirstUI')
```
Comment (meelie) [inline src/js/yeldaWebchatInjector.js:177]: no need to repeat this comment, it's explained just above

File: src/js/yeldaWebchatInjector.js [hunk 5]
```diff
@@ -158,25 +158,43 @@ class YeldaChat {
...
+        this.openChat()
         return
-      }
+      } else {
```
Comment (meelie) [inline src/js/yeldaWebchatInjector.js:180]: no need of else here, has we return in the if

File: src/js/yeldaWebchatInjector.js (hunk 4 above)
Comment (meelie) [inline src/js/yeldaWebchatInjector.js:177]: Can you add the needed test for this please ?

- if isVoiceFirstUI is true, voiceFirstUI is added to the iframeContainer
- if isVoiceFirstUI is false, voiceFirstUI is not added to the iframeContainer

- if isVoiceFirstUI is true, assistantImage is not added to the webChatContainer
- if isVoiceFirstUI is false, assistantImage is added to the webChatContainer

**Review by meelie — CHANGES_REQUESTED** [review]

File: test/test.js [hunk 6]
```diff
@@ -180,19 +175,65 @@ describe('YeldaChat', () => {
...
+        'assistantId': '12345678',
+        'chatPath': 'chat'
+      }
+
+      yeldaChat.setupChat(mockData)
+      // setTimeout used to wait the for the mock response
+      setTimeout(() => {
+        expect(yeldaChat.iframeContainer).to.have.attribute('class', 'yelda_iframe_container voiceFirstUI y_active')
+      })
+    })
```
Comment (meelie) [inline test/test.js:202]: Are you sure this test is working ?
I commented `        // this.iframeContainer.classList.add('voiceFirstUI')` and the test still pass whereas it should fail.
image

File: test/test.js [hunk 7]
```diff
@@ -180,19 +175,65 @@ describe('YeldaChat', () => {
...
+
+    it('expect assistantImage not exists in document', () => {
+      expect(yeldaChat.assistantImage).to.be.null
+    })
+
+    it('should not contain assistant image', () => {
+      expect(document.querySelector('#yelda_assistant_img')).to.be.null
+    })
```
Comment (meelie) [inline test/test.js:210]: Same here : whatever i do, the test pass whereas it should not
```
// test passed
 if (isVoiceFirstUI) {
        this.openChat()
        this.webChatContainer.appendChild(this.assistantImage)
        return
      }
// test passed
 if (isVoiceFirstUI) {
        this.iframeContainer.classList.add('voiceFirstUI')
        this.openChat()
        this.assistantImage = 'ntull'
        return
      }
```

Can you recheck ?

File: test/test.js [hunk 8]
```diff
@@ -180,19 +175,65 @@ describe('YeldaChat', () => {
...
+
+  })
+
   describe('yeldaChat.createWebChatURL', () => {
     it('should return valid iframe url', () => {
+      // resets the previous isVoiceFirstUI response
+      mock.reset()
+      mock.get('https://app.yelda.ai//assistants/12345678/chatBubble/fr_FR', {
```
Comment (meelie) [inline test/test.js:218]: - use a const at the top ?
- https://app.yelda.ai/assistants/12345678/chatBubble/fr_FR   (one slash)

**Review by meelie — CHANGES_REQUESTED** [review]
tests to recheck 
thanks !

File: test/test.js (hunk 6 above)
Comment (vishnu-ravi) [reply test/test.js:202]: Are you sure, you build the code before running the test command ?
npm run prepack will do both build and test

Comment (meelie) [reply test/test.js]: haha indeed, i was sure we were testing the injector directly, not the build one.
OK for me, thanks

**Review by meelie — APPROVED** [review]
Thanks !

**Pull request merged**